simulations/vip-178/vip-178/simulations.ts

```diff
@@ -0,0 +1,320 @@
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { expect } from "chai";
+import { BigNumberish } from "ethers";
+import { Contract } from "ethers";
+import { parseUnits } from "ethers/lib/utils";
+import { ethers } from "hardhat";
+
+import { expectEvents, initMainnetUser } from "../../../src/utils";
+import { forking, testVip } from "../../../src/vip-framework";
+import { vip178 } from "../../../vips/vip-178/vip-178";
+import COMPTROLLER_ABI from "./abi/comptroller.json";
+import ERC20_ABI from "./abi/erc20.json";
+import POOL_REGISTRY_ABI from "./abi/poolRegistry.json";
+import RATE_MODEL_ABI from "./abi/rateModel.json";
+import REWARD_DISTRIBUTOR_ABI from "./abi/rewardsDistributor.json";
+import VTOKEN_ABI from "./abi/vToken.json";
+
+const agEUR = "0x12f31B73D812C6Bb0d735a218c086d44D5fe5f89";
+const POOL_REGISTRY = "0x9F7b01A536aFA00EF10310A162877fd792cD0666";
+const VTOKEN_RECEIVER_agEUR = "0xc444949e0054a23c44fc45789738bdf64aed2391";
+const VagEUR_Stablecoins = "0x795DE779Be00Ea46eA97a28BDD38d9ED570BCF0F";
+const STABLECOIN_COMPTROLLER = "0x94c1495cD4c557f1560Cbd68EAB0d197e6291571";
+const NORMAL_TIMELOCK = "0x939bD8d64c0A9583A7Dcea9933f7b21697ab6396";
+const REWARD_DISTRIBUTOR = "0x177ED4625F57cEa2804EA3A396c8Ff78f314F1CA";
+const ANGLE = "0x97B6897AAd7aBa3861c04C0e6388Fc02AF1F227f";
+const PROTOCOL_SHARE_RESERVE = "0xfB5bE09a1FA6CFDA075aB1E69FE83ce8324682e4";
+const SHORTFALL = "0xf37530A8a810Fcb501AA0Ecd0B0699388F0F2209";
+
+forking(32162545, () => {
+  let poolRegistry: Contract;
+  let comptroller: Contract;
+  let vagEUR: Contract;
+  let rewardsDistributor: Contract;
+
+  before(async () => {
+    poolRegistry = await ethers.getContractAt(POOL_REGISTRY_ABI, POOL_REGISTRY);
+    comptroller = await ethers.getContractAt(COMPTROLLER_ABI, STABLECOIN_COMPTROLLER);
+    vagEUR = await ethers.getContractAt(VTOKEN_ABI, VagEUR_Stablecoins);
+    rewardsDistributor = await ethers.getContractAt(REWARD_DISTRIBUTOR_ABI, REWARD_DISTRIBUTOR);
+  });
+
+  describe("Contracts setup", () => {
+    const checkVToken = (
```

Comment:
We could move this check function to this folder, to reuse it in the future: https://github.com/VenusProtocol/vips/blob/develop/src/vip-framework/checks